### PR TITLE
Processes get added to db in one big query instead of individual inserts.

### PR DESCRIPTION
**Pull request by spinza**
When polling unix agent processes we add the processes to the db in one large query instead of an individual query for each of them.  Reduces the time taken.

### CODE REVIEW

File: includes/polling/unix-agent.inc.php [hunk 1]
```diff
@@ -95,14 +95,22 @@
     {
       echo("Processes: ");
       dbDelete('processes', 'device_id = ?', array($device['device_id']));
+      $query = "INSERT INTO `processes` (`device_id`,`pid`,`user`,`vsz`,`rss`,`cputime`,`command`)  VALUES ";
+      $values = "";
       foreach (explode("\n", $agent_data['ps']) as $process)
       {
         $process = preg_replace("/\((.*),([0-9]*),([0-9]*),([0-9\:]*),([0-9]*)\)\ (.*)/", "\\1|\\2|\\3|\\4|\\5|\\6", $process);
```
Comment (spinza) [inline includes/polling/unix-agent.inc.php]: @laf this does the escaping?

**Comment by spinza** [comment]
All escaped.

**Comment by paulgear** [comment]
This looks OK to me, although I wonder whether we wouldn't get the same performance by simply wrapping this in a begin/commit block.

**Comment by spinza** [comment]
@paulgear are you wondering or do you know?  It would be good if it worked as I was going to tackle the package code next as it is similarly using individual inserts and deletes. Though that code is a bit more complex.

Just found this: https://dev.mysql.com/doc/refman/5.7/en/insert-speed.html
Not sure if the transaction will help as it is still multiple connections or have I got it upside down?

**Comment by spinza** [comment]
I actually should write a dbFacile dbInsertBulk, dbDeleteBulk and dbUpdateBulk functions for this.  These could take table name, field headers, primary key and a array for the data as parameters?  We could then optimise there?
Next PR?

**Comment by spinza** [comment]
Rebased.

**Pull request merged**